### PR TITLE
Sprint 19

### PR DESCRIPTION
**Pull request by ericsorenson**
# Sprint 19

## [530 Docket Clerk: Service on Parties of Signed Stipulated Decision](https://github.com/flexion/ef-cms/issues/530)
image

## [1566 External User: View documents uploads before they are associated with case/docket entry (Orphan Documents)](https://github.com/flexion/ef-cms/issues/1566)
image

## [1448 Feedback from Practitioner Interviews](https://github.com/flexion/ef-cms/issues/1448)
image

## [1299 Petitions Clerk: New Flow for Paper Filed Petition](https://github.com/flexion/ef-cms/issues/1299)
image

## [49 CMS User: View and print case record](https://github.com/flexion/ef-cms/issues/49)
image

## [1508 Usability Testing: File a Document Form Clean-Up](https://github.com/flexion/ef-cms/issues/1508)
image

### CODE REVIEW

**Comment by waldoj** [comment]
Here's the review checklist that I'll work my way through in reviewing this PR:

- [ ] begin by deleting any existing copy of the Git repository and cloning a fresh copy, running `docker system prune` to remove any cached artifacts from any prior code review, and running `./clean-up.sh` to update Node packages
- [ ] review the pull request itself, to get oriented
	- [ ] read the description of the pull request, which should summarize the changes made
	- [ ] read through every task on the Scrum board that's encompassed by this pull request
	- [ ] read the description of the commits that comprise the pull request
- [ ] stand up the site locally
	- [ ] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
		- [ ] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
		- [ ] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/), including Internet Explorer 11
		- [ ] test in Mobile Safari and Mobile Chrome, with the caveat that not all internal Court functionality will be necessary on these platforms
	- [ ] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [ ] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [ ] review for accessibility
		- [ ] use an automated audit tool, such as Chrome Audit or aXe
		- [ ] navigate site only with the keyboard
		- [ ] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [ ] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [ ] review static code analysis results in SonarCloud
- [ ] run a security audit of dependencies (`npm audit`) in `efcms-service`, `web-client`, and `shared`, to ensure that there are no vulnerabilities that will be deployed to production (as opposed to vulnerabilities that only have an impact on the development environment)
- [ ] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [ ] skim all new code, in the context of existing code, [looking for problems](https://github.com/ustaxcourt/ef-cms/blob/staging/docs/CODE_REVIEW.md#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [ ] review all tests
	- [ ] look at code coverage of tests in SonarCloud
	- [ ] methodically review all new tests for correctness, quality of naming
- [ ] determine what code isn’t tested, review that rigorously
- [ ] review documentation to ensure that it matches changes
- [ ] provide comments on the pull request on GitHub, as necessary
	- [ ] for comments that are specific to a particular line of code, comment on those specific lines
	- [ ] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [ ] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

File: web-client/src/presenter/actions/rescanBatchAction.test.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    expect(result.state.isScanning).toBeTruthy();
+    expect(mockStartScanSession).toHaveBeenCalled();
+    expect(result.state.batches.petition[1].pages).toEqual([
+      { e: 5 },
+      { f: 6 },
+    ]);
+  });
+
+  it('tells the TWAIN library to begin image aquisition with no scanning device set', async () => {
```
[the remark this answers is not in the document]
Comment (rachaelparris) [reply web-client/src/presenter/actions/rescanBatchAction.test.js]: Addressed

File: web-client/integration-tests/signAndServeStipulatedDecision.test.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        },
+      },
+    };
+  });
+
+  let docketNumber = null;
+  let signedDocumentId = null;
+  let caseDetail;
+
+  it('login as a tax payer and create a case', async () => {
```
[the remark this answers is not in the document]
Comment (rachaelparris) [reply web-client/integration-tests/signAndServeStipulatedDecision.test.js]: Addressed

File: shared/src/dispatchers/ses/sendBulkTemplatedEmail.test.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        },
+      ],
+      templateName: 'case_served',
+    });
+
+    expect(sendBulkTemplatedEmailStub.getCall(0).args[0]).toMatchObject({
+      Destinations: [
+        {
+          Destination: {
+            ToAddresses: ['test.email@gmail.com'],
```
[the remark this answers is not in the document]
Comment (rachaelparris) [reply shared/src/dispatchers/ses/sendBulkTemplatedEmail.test.js]: Addressed

File: shared/src/dispatchers/ses/sendBulkTemplatedEmail.test.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      logger: {
+        error: () => null,
+      },
+    };
+
+    await sendBulkTemplatedEmail({
+      applicationContext,
+      destinations: [
+        {
+          email: 'test.email@gmail.com',
```
[the remark this answers is not in the document]
Comment (rachaelparris) [reply shared/src/dispatchers/ses/sendBulkTemplatedEmail.test.js]: Addressed

File: shared/src/business/useCases/updatePrimaryContactInteractor.test.js [hunk 5]
```diff
@@ -27,12 +28,46 @@ const applicationContext = {
   },
 };
 
-describe('Updates contactPrimary on the given case', () => {
+describe('updatePrimaryContactInteractor', () => {
```
[the remark this answers is not in the document]
Comment (rachaelparris) [reply shared/src/business/useCases/updatePrimaryContactInteractor.test.js]: Addressed

File: shared/src/business/useCases/serveSignedStipDecisionInteractor.test.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    await serveSignedStipDecisionInteractor({
+      applicationContext,
+      caseId: MOCK_CASE.caseId,
+      documentId: '1805d1ab-18d0-43ec-bafb-654e83405416',
+    });
+
+    expect(updatedCase.documents[0].servedParties).toHaveLength(3);
+  });
+
+  it('throws an unathorized error when a non docketclerk role attempts to serve', async () => {
```
[the remark this answers is not in the document]
Comment (rachaelparris) [reply shared/src/business/useCases/serveSignedStipDecisionInteractor.test.js]: Addressed

File: shared/src/business/useCases/serveSignedStipDecisionInteractor.test.js [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+      await serveSignedStipDecisionInteractor({
+        applicationContext,
+        caseId: MOCK_CASE.caseId,
+        documentId: '1805d1ab-18d0-43ec-bafb-654e83405416',
+      });
+    } catch (err) {
+      error = err;
+    }
+
+    expect(error).toBeDefined();
```
[the remark this answers is not in the document]
Comment (rachaelparris) [reply shared/src/business/useCases/serveSignedStipDecisionInteractor.test.js]: Addressed

File: shared/src/business/entities/externalDocument/externalDocumentHelpers.test.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+      };
+      const schemaOptionalItems = {
+        somethingElse: joi.string(),
+      };
+      makeRequiredHelper({
+        itemName: 'somethingElse2',
+        schema,
+        schemaOptionalItems,
+      });
+      expect(schema.somethingElse2).toEqual(undefined);
```
[the remark this answers is not in the document]
Comment (rachaelparris) [reply shared/src/business/entities/externalDocument/externalDocumentHelpers.test.js]: Addressed

File: shared/src/business/entities/Scan.test.js [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+      rawBatch: {},
+    });
+
+    scan.addBatch(batch);
+    scan.removeBatch(batch);
+
+    expect(scan.batches).toHaveLength(0);
+  });
+
+  it('returns an array aggregated pages from all batches', () => {
```
[the remark this answers is not in the document]
Comment (rachaelparris) [reply shared/src/business/entities/Scan.test.js]: Addressed
